addons/mail/static/src/discuss/core/common/discuss_channel_model.js

```diff
@@ -332,6 +333,27 @@ export class DiscussChannel extends Record {
         }
     }
 
+    async leaveChannel() {
+        if (this.channel_type !== "group" && this.create_uid?.eq(this.store.self_user)) {
+            await this.askLeaveConfirmation(
+                _t("You are the administrator of this channel. Are you sure you want to leave?")
+            );
+        }
+        if (this.channel_type === "group") {
+            await this.askLeaveConfirmation(
+                _t(
+                    "You are about to leave this group conversation and will no longer have access to it unless you are invited again. Are you sure you want to continue?"
+                )
+            );
+        }
+        await this.thread.closeChatWindow();
```

Comment:
Could just `this.closeChatWindow()` and let code of inheritance pick the method on `thread`.